spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/SparkFilters.java

```diff
@@ -178,6 +178,7 @@ public static Expression convert(Filter filter) {
             In childInFilter = (In) childFilter;
             Expression notIn = notIn(unquote(childInFilter.attribute()),
                 Stream.of(childInFilter.values())
+                    .filter(Objects::nonNull)
```

Comment:
This refers to the process in L166. I think this is because we don't support NULL literal in Iceberg. However,  I think here have changed the original filter meanings. @rdblue, should we support NULL literal in Iceberg?

Reply:
> should we support NULL literal in Iceberg?

No. `null` is purposely not allowed as a literal because that brings in differences between SQL 3-valued boolean semantics and normal semantics. Engines are responsible for interpreting and rewriting filters. Similarly, in cases like this we actually want to reject null values in IN predicates. The user or the engine needs to rewrite before passing to Iceberg.